pkg/network/migration/evaluator.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package migration
+
+import (
+	"time"
+
+	k8scorev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/network/vmispec"
+)
+
+const DynamicNetworkControllerGracePeriod = 15 * time.Second
+
+type migrationRequirementKind int
+
+const (
+	notRequired migrationRequirementKind = iota
+	immediateMigration
+	pendingMigration
+)
+
+type timeProviderFunc func() metav1.Time
+
+type Evaluator struct {
+	timeProvider timeProviderFunc
+}
+
+func NewEvaluator() Evaluator {
+	return NewEvaluatorWithTimeProvider(metav1.Now)
+}
+
+func NewEvaluatorWithTimeProvider(timeProvider timeProviderFunc) Evaluator {
+	return Evaluator{timeProvider: timeProvider}
+}
+
+func (e Evaluator) Evaluate(vmi *v1.VirtualMachineInstance) k8scorev1.ConditionStatus {
+	result := shouldVMIBeMarkedForAutoMigration(
+		vmi.Spec.Domain.Devices.Interfaces,
+		vmi.Spec.Networks,
+		vmi.Status.Interfaces,
```

Comment:
Are we sure that all of these fields can't be `nil` at this point?

Reply:
The code is expected to work even if all three slices are `nil`.
There is a unit test for it in `pkg/network/migration/evaluator_test.go`:
> Evaluator Should not require migration when no networks are specified.

Reply:
Thanks, you're right